### PR TITLE
Extracted DateField from boundDateField

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
None
#### What's this PR do?
Creates `DateField` which is not specific to the profile code. I will use this in the UI work regarding #1115

#### How should this be manually tested?
Edit your birthdate and a graduation date in the profile.

### CODE REVIEW

File: static/js/util/profile_edit.js [hunk 1]
```diff
@@ -147,146 +137,15 @@ export function boundDateField(keySet: string[], label: string, omitDay: boolean
...
-      }}
-      fullWidth={true}
-      value={edit.year !== undefined ? edit.year : ""}
-      onChange={e => setNewDate(undefined, undefined, e.target.value)}
-    />
-  </div>;
+  return <DateField
+    data={profile}
+    errors={errors}
+    update={updateProfile}
```
Comment (gsidebo) [inline static/js/util/profile_edit.js]: would `updateFunc` or `updateHandler` be a better name here? `update` is ambiguous and kind of confusing

File: static/js/util/profile_edit.js [hunk 2]
```diff
@@ -147,146 +137,15 @@ export function boundDateField(keySet: string[], label: string, omitDay: boolean
...
-      style={{
-        maxWidth: "4em"
-      }}
-      fullWidth={true}
-      value={edit.year !== undefined ? edit.year : ""}
-      onChange={e => setNewDate(undefined, undefined, e.target.value)}
-    />
-  </div>;
+  return <DateField
+    data={profile}
```
Comment (gsidebo) [inline static/js/util/profile_edit.js]: not wild about a prop as ambiguous as `data`, but truth be told, i don't think it was any more sensible before this refactor. we can look into this post-deadline

**Comment by gsidebo** [comment]
just one small request

File: static/js/util/profile_edit.js (hunk 1 above)
Comment (noisecapella) [inline static/js/util/profile_edit.js]: Sure, I'll switch to `updateHandler`

**Comment by noisecapella** [comment]
Made the change

**Comment by gsidebo** [comment]
looks good 👍

**Pull request merged**